core/drand_daemon.go

```diff
@@ -152,27 +155,50 @@ func (dd *DrandDaemon) InstantiateBeaconProcess(beaconID string, store key.Store
 	}
 
 	dd.state.Lock()
-	dd.beaconProcesses[beaconID] = bp
+	dd.bpByID[beaconID] = bp
 	dd.state.Unlock()
 
 	return bp, nil
 }
 
+func (dd *DrandDaemon) AddNewChainHash(beaconID string, bp *BeaconProcess) {
+	dd.state.Lock()
+	chainHash := chain.NewChainInfo(bp.group).HashString()
```

Comment:
can we see if there is already a beacon set for the hash, and if the beaconID has changed log that the beacon ID we're mapping from a hash is changing. That should never happen, so worth having some visibility on the expected invariant.